src/namerd.rs

```diff
@@ -53,8 +54,8 @@ pub fn resolve<C>(addr: net::SocketAddr,
 {
     let url = {
         let base = format!("http://{}:{}/api/1/resolve/{}",
-                           addr.ip(),
-                           addr.port().to_string(),
+                           host,
```

Comment:
I think this will result in a blocking resolution? I can find out via strace. If so, we can use https://github.com/alexcrichton/futures-rs/tree/master/futures-cpupool